klayout/drc/rule_decks/main.drc

```diff
@@ -0,0 +1,1072 @@
+
+mcell_feol_mk  = get_polygons(11 , 17)
+count   = mcell_feol_mk.count()
+logger.info("mcell_feol_mk has %d polygons" % [count])
+polygons_count  += count
+
+ymtp_mk        = get_polygons(86 , 17)
+count   = ymtp_mk.count()
+logger.info("ymtp_mk has %d polygons" % [count])
+polygons_count  += count
+
+dev_wf_mk      = get_polygons(128, 17)
+count   = dev_wf_mk.count()
+logger.info("dev_wf_mk has %d polygons" % [count])
+polygons_count  += count
+
+comp_label     = get_polygons(22 , 10)
+count   = comp_label.count()
+logger.info("comp_label has %d polygons" % [count])
+polygons_count  += count
+
+poly2_label    = get_polygons(30 , 10)
+count   = poly2_label.count()
+logger.info("poly2_label has %d polygons" % [count])
+polygons_count  += count
+
+mdiode         = get_polygons(116, 5 )
+count   = mdiode.count()
+logger.info("mdiode has %d polygons" % [count])
+polygons_count  += count
+
+contact        = get_polygons(33 , 0 )
+count   = contact.count()
+logger.info("contact has %d polygons" % [count])
+polygons_count  += count
+
+metal1_drawn   = get_polygons(34 , 0 )
+count   = metal1_drawn.count()
+logger.info("metal1_drawn has %d polygons" % [count])
+polygons_count  += count
+
+metal1_dummy   = get_polygons(34 , 4 )
+count   = metal1_dummy.count()
+logger.info("metal1_dummy has %d polygons" % [count])
+polygons_count  += count
+
+metal1         = metal1_drawn + metal1_dummy
+
+metal1_label   = get_polygons(34 , 10)
+count   = metal1_label.count()
+logger.info("metal1_label has %d polygons" % [count])
+polygons_count  += count
+
+metal1_slot    = get_polygons(34 , 3 )
+count   = metal1_slot.count()
+logger.info("metal1_slot has %d polygons" % [count])
+polygons_count  += count
+
+metal1_blk     = get_polygons(34 , 5 )
+count   = metal1_blk.count()
+logger.info("metal1_blk has %d polygons" % [count])
+polygons_count  += count
+
+via1           = get_polygons(35 , 0 )
+count   = via1.count()
+logger.info("via1 has %d polygons" % [count])
+polygons_count  += count
+
+
+if METAL_LEVEL == "2LM"
+  metal2_drawn  = get_polygons(36 , 0 )
+  count   = metal2_drawn.count()
+  logger.info("metal2_drawn has %d polygons" % [count])
+  polygons_count  += count
+
+  metal2_dummy  = get_polygons(36 , 4 )
+  count   = metal2_dummy.count()
+  logger.info("metal2_dummy has %d polygons" % [count])
+  polygons_count  += count
+
+  metal2        = metal2_drawn + metal2_drawn
+
+  metal2_label  = get_polygons(36 , 10)
+  count   = metal2_label.count()
+  logger.info("metal2_label has %d polygons" % [count])
+  polygons_count  += count
+
+  metal2_slot   = get_polygons(36 , 3 )
+  count   = metal2_slot.count()
+  logger.info("metal2_slot has %d polygons" % [count])
+  polygons_count  += count
+
+  metal2_blk    = get_polygons(36 , 5 )
+  count   = metal2_blk.count()
+  logger.info("metal2_blk has %d polygons" % [count])
+  polygons_count  += count
+  
+  top_via       = via1
+  topmin1_via   = contact
+  top_metal     = metal2
+  topmin1_metal = metal1
+  
+else
```

Comment:
I think we can get rid of the elses if we don't assign the `top*` variables

Reply:
No we can't get rid of them as they are used later in the TopMetal rules.

Reply:
maybe a better way to structure that code would be with a table and a for loop? (since you mostly repeat the same pattern across all the if/else block)

Reply:
I already opened an issue for that.